ext/mbstring/mbstring.c

```diff
@@ -2183,6 +2190,196 @@ PHP_FUNCTION(mb_output_handler)
+
+    /*
+     * +----------------------------------------------------------------------+
+     * | first scenario:    1,2,4-bytes fixed width encodings (head part)     |
+     * +----------------------------------------------------------------------+
+     */
+    if (mbfl_encoding->flag & MBFL_ENCTYPE_SBCS) { /* 1 byte */
+        mb_len = string.len;
+        chunk_len = (size_t)split_length; /* chunk length in bytes */
+
+    } else if (mbfl_encoding->flag & (MBFL_ENCTYPE_WCS2BE | MBFL_ENCTYPE_WCS2LE )) { /* 2 bytes */
+        mb_len = string.len>>1; /* eq. string.len / 2 String length in chars */
+        chunk_len = split_length<<1; /* eq. split_length * 2 chunk length in bytes */
+
+    } else if (mbfl_encoding->flag & (MBFL_ENCTYPE_WCS4BE | MBFL_ENCTYPE_WCS4LE )) { /* 4 bytes */
+        mb_len = string.len>>2; /* eq. string.len / 4 */
+        chunk_len = split_length<<2; /* eq. split_length * 4 */
+    /*
+     * +----------------------------------------------------------------------+
+     * | second scenario: "variable width encodings with length table"        |
+     * +----------------------------------------------------------------------+
+     */
+    } else if (mbfl_encoding->mblen_table != NULL) {
+        char unsigned const *mbtab = mbfl_encoding->mblen_table;
+
+        /* assume that we have 2-bytes characters */
+        array_init_size(return_value, (string.len>>1 + split_length) / split_length);
+
+        while (p < last) { /* split cycle work until the cursor has reached the last byte */
+            char const *chunk_p = p; /* chunk first byte pointer */
+            chunk_len = 0; /* chunk length in bytes */
+            for (uint32_t char_count = 0; char_count < split_length; ++char_count) {
+                char unsigned const m = mbtab[*p]; /* single character length table */
+                chunk_len += m;
+                p += m;
+            }
+            if (p > last) chunk_len -= p - last; /* check if chunk is in bounds */
+            add_next_index_stringl(return_value, chunk_p, chunk_len);
+        }
+        return;
+
+    /*
+     * +----------------------------------------------------------------------+
+     * | third scenario: "else multibyte encodings"                           |
+     * +----------------------------------------------------------------------+
+     */
+    } else {
+        mbfl_convert_filter *filter, *decoder;
+
+        /* assume that we have 2-bytes characters */
+        array_init_size(return_value, (string.len>>1 + split_length) / split_length);
+        p = string.val; /* reset string cursor position */
+
+        /* decoder filter to decode wchar to encoding */
+        mbfl_memory_device device;
+        mbfl_memory_device_init(&device, split_length + 1, 0);
+
+        decoder = mbfl_convert_filter_new(
+                &mbfl_encoding_wchar,
+                string.encoding,
+                mbfl_memory_device_output,
+                0,
```

Comment:
This should be NULL to make clear it's a pointer argument.

Reply:
Fixed.